gui/js/lastfiles.js

```diff
@@ -5,15 +5,22 @@
 
 const fs = require('fs')
 const path = require('path')
+const async = require('async')
 
 let lastFilesPath = ''
 let lastFiles = Promise.resolve([])
+let persistQueue = null
 
 const log = require('../../core/app').logger({
   component: 'GUI'
 })
 
-module.exports.init = desktop => {
+const writeJSON = (data, callback) => {
+  fs.writeFile(lastFilesPath, data, callback)
+}
+
+const init = desktop => {
+  persistQueue = Promise.promisifyAll(async.queue(writeJSON))
```

Comment:
I'm wondering how it works since you're not returning the Promise in writeJSON ? :thinking:

Reply:
`writeJSON` passes the callback to `fs.writeFile` so the promise is resolved (or rejected) by `writeFile`?!

Reply:
How the promise is resolved or rejected is not my the point. 

Here `writeJSON` return `undefined` so how promisifyAll can know about waiting a promise?

Reply:
That tripped me up too, but `async.queue` takes a function as param, and the function receives a node style `callback` that you call when you're done. Here we pass it to `writeFile` and it handles it from there ;)